test/azure/src/main/java/fixtures/azureparametergrouping/implementation/ParameterGroupingsImpl.java

```diff
@@ -150,30 +165,43 @@ public void postOptional() {
      * @throws IllegalArgumentException thrown if parameters fail the validation
      * @return the {@link ServiceFuture} object
      */
-    public ServiceFuture<Void> postOptionalAsync(final ServiceCallback<Void> serviceCallback) {
+    public ServiceFuture<Void> postOptionalAsync(ServiceCallback<Void> serviceCallback) {
         return ServiceFuture.fromBody(postOptionalAsync(), serviceCallback);
     }
 
     /**
      * Post a bunch of optional parameters grouped.
      *
      * @throws IllegalArgumentException thrown if parameters fail the validation
-     * @return the {@link Void} object if successful.
+     * @return a {@link Single} emitting the RestResponse<Void, Void> object
      */
-    public Single<Void> postOptionalAsync() {
+    public Single<RestResponse<Void, Void>> postOptionalWithRestResponseAsync() {
         final ParameterGroupingPostOptionalParameters parameterGroupingPostOptionalParameters = null;
-        String customHeader = null;
-        Integer query = null;
+        Validator.validate(parameterGroupingPostOptionalParameters);
+    String customHeader = null;
+    Integer query = null;
         return service.postOptional(this.client.acceptLanguage(), customHeader, query, this.client.userAgent());
     }
 
     /**
      * Post a bunch of optional parameters grouped.
      *
+     * @throws IllegalArgumentException thrown if parameters fail the validation
```

Comment:
This function has no parameters. What parameters is the comment talking about?

Reply:
This means an if (!parameters.IsEmpty) check needs to be added somewhere as part of a refactor.